src/containers/Trade.svelte

```diff
@@ -180,38 +186,24 @@
     showTradeModal = true;
 
     try {
-      const utxos = await window.marina.getCoins();
-      const trade = new Trade({
-        providerUrl: provider.endpoint,
-        explorerUrl: 'https://blockstream.info/liquid/api',
-        coinSelector: greedyCoinSelector(),
-        utxos: utxos.filter((u) => (u as UtxoInterface).prevout),
-      });
-
-      const { hash, precision } = CoinToAssetByChain['liquid'][sendCoin];
-      const amountToBeSentInSatoshis = toSatoshi(sendAmount, precision);
-
-      const isBuy = hash === provider.market.quoteAsset;
-
+      const amountToBeSentInSatoshis = toSatoshi(
+        sendAmount,
+        pair.send.precision
+      );
       console.debug(
-        `trading ${amountToBeSentInSatoshis} of ${sendCoin} for ${receiveCoin}...`
+        `trading ${amountToBeSentInSatoshis} of ${pair.send.ticker} for ${pair.receive.ticker}...`
       );
 
-      if (isBuy) {
-        txid = await trade.buy({
-          market: provider.market,
-          amount: amountToBeSentInSatoshis.toNumber(),
-          asset: hash,
-          identity,
-        });
-      } else {
-        txid = await trade.sell({
-          market: provider.market,
-          amount: amountToBeSentInSatoshis.toNumber(),
-          asset: hash,
-          identity,
-        });
-      }
+      txid = await makeTrade(
+        identity,
+        {
+          sats: amountToBeSentInSatoshis.toNumber(),
+          asset: pair.send.assetHash,
+        },
+        bestOrder,
+        $marinaStore.utxos,
+        EXPLORER
+      );
 
       tradeStatus = TradeStatus.COMPLETED;
     } catch (e) {
```

Comment:
There is an issue: are losing specificity on the error shown to user.

If the user has no available utxo, previously the error shown was "You need at least one utxo to trade", which is good, but now the error shown to the user is "error while making trade", which is worst.

The same thing occurs when there are no orders available.